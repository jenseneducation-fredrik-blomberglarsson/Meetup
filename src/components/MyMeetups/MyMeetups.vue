<template>
  <div class="myMeetups"></div>
</template>

<script>
export default {};
</script>

<style>
</style>