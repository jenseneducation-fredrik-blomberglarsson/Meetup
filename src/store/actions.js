import { getMeetups } from "../data/meetupData";

const actions = {
  async fetchMeetups(context) {
    const data = await getMeetups();
    context.commit("setMeetups", data);
  },
  addThisMeetup(context, meetup) {
    context.commit('addMeetup', meetup);
  },
  removeThisMeetup(context, id) {
    context.commit('removeMeetup', id)
  },
  switchThisMeetupToggle(context, boolean) {
    context.commit('switchMeetupToggle', boolean);
  }



  /*   isBtnVisible(context) {
      context.commit('isVisible');
    } */
}

export default actions;