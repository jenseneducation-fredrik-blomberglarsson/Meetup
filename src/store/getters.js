const getters = {
  getMeetups: state => {
    return state.meetups;
  },
  getJoinedMeetups: state => {
    return state.joinedMeetups;
  },
  getMeetupToggle: state => {
    return state.meetupToggle;
  }
}


export default getters;