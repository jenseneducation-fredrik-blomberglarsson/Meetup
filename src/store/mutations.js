const mutations = {
  setMeetups(state, meetups) {
    state.meetups = meetups;
  }
}

export default mutations;