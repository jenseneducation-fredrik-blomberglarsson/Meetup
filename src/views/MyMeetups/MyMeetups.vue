<template>
  <div class="myMeetups">
    <h1>Attending meetups</h1>
    <MyMeetupList />
  </div>
</template>

<script>
import MyMeetupList from "@/components/MyMeetupList/MyMeetupList.vue";
export default {
  name: "MyMeetups",
  components: { MyMeetupList },
};
</script>

<style>
</style>