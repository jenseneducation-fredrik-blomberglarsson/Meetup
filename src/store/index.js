import Vue from 'vue'
import Vuex from 'vuex'
import actions from './actions.js';
import mutations from './mutations.js';
import getters from './getters.js';

Vue.use(Vuex)

//Global state
export default new Vuex.Store({
  state: {
    meetups: [],
    joinedMeetups: [],
    previousMeetups: [],
    meetupToggle: false,
  },

  actions: actions,
  mutations: mutations,
  getters: getters
});