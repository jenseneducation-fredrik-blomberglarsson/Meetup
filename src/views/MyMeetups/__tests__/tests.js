import { shallowMount, mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import store from '../../../store/index.js';
import MyMeetups from '@/views/MyMeetups/MyMeetups.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('The page is rendered', () => {
  test('Test so that everything renders properly', async () => {
    const wrapper = shallowMount(MyMeetups, { store, localVue });
    expect(wrapper.element).toMatchSnapshot();
  })
})