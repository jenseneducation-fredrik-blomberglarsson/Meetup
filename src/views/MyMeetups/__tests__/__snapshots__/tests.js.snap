// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The page is rendered Test so that everything renders properly 1`] = `
<div
  class="myMeetups"
>
  <h1>
    Attending meetups
  </h1>
   
  <mymeetuplist-stub />
</div>
`;
